enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -0,0 +1,806 @@
+        * Using a NodePort-type service is not supported (but the cluster admin could define one manually).
+        * Using host networking is not supported.
+    * This enhancement does not monitor Envoy using a canary application.
+    * None of the following items can be configured on Envoy at this time:
+        * HTTP error pages.
+        * Access logs.
+        * Pod placement.
+        * Allowed TLS version or ciphers.
+        * HTTP compression.
+        * Client TLS.
+        * HTTP `forwarded` header or other HTTP headers.
+        * Envoy performance tuning options.
+* OpenShift does not provide special support for third-party (non-Red Hat) Gateway API implementations.
+    * OpenShift does not configure LB, DNS, etc. for third-party implementations.
+* OpenShift does not support using Istio APIs to configure ingress (but they may still be used to configure mesh).
+    * If an Istio feature cannot be configured using Gateway API, that feature is not supported.
+* This enhancement does not deprecate OpenShift router or the Route API.
+    * OpenShift router remains a standard OpenShift component.
+    * Istio/Envoy do not watch Routes or Ingresses.
+    * OpenShift does not translate Routes or Ingresses into HTTPRoutes.
+
+## Proposal
+
+Most of the proposed changes are to Ingress Operator.  Some API changes may also
+be warranted.
+
+### Changes to the Ingress Operator
+
+With this enhancement, the Ingress Operator is responsible for managing Gateway
+API CRDs, installing OpenShift Service Mesh Operator, managing a
+LoadBalancer-type service for a Gateway, and managing DNS for the same.  This
+requires the following changes:
+
+* Refactor the existing service and DNS management logic to enable its re-use.
+* Modify the DNS management logic not to require an associated IngressController.
+* Add a new controller to manage Gateway API CRDs and install and configure Service Mesh Operator.
+* Add a new controller to manage DNS for Gateways.
+
+These changes are explained in more detail in the following sections.
+
+#### Refactoring Service and DNS Management
+
+The Ingress Operator is designed around the IngressController API, and much
+logic assumes that there is some IngressController object that specifies the
+relevant configuration and reports status.  In particular, the operator has
+logic for reconciling services and DNS records for an IngressController, which
+specifies the desired service type (NodePort, LoadBalancer, or none) and the
+desired DNS domain.  This code needs to be refactored so that it can be re-used
+by logic that reconciles Gateway objects.
+
+The initial enhancement provides a subset of the options that are available with
+the IngressController API.  In particular, using a LoadBalancer-type service is
+supported; using a NodePort-type service or host networking is not.
+
+#### DNS Management Logic
```

Comment:
*nit* The placement of this `#### DNS Management Logic` paragraph feels a little awkward to me, should it be combined, or adjacent to `#### Controller to manage DNS records for Gateway listeners`? Aren't they doesn't `DNS Management Logic` feed into `Controller to manage DNS records for Gateway listeners`?

Reply:
I'll make those sections adjacent and modify the title of the section about the DNS controller to make it clearer that it is specific to that controller.  I prefer to keep the two sections distinct (but adjacent) because one describes changes to an existing controller and the other describes the design of a new controller.